onnxruntime/core/providers/coreml/builders/impl/argmax_op_builder.cc

```diff
@@ -32,57 +32,40 @@ Status ArgMaxOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
 
   NodeAttrHelper helper(node);
   const auto axis = helper.Get("axis", 0);
-  const auto keep_dims = helper.Get("keep_dims", 1);
-  const bool removedim = keep_dims != 1;
+  const auto keepdims = helper.Get("keepdims", 1);
+  const bool removedim = keepdims != 1;
 
   auto* coreml_argmax = layer->mutable_argmax();
   coreml_argmax->set_axis(axis);
   coreml_argmax->set_removedim(removedim);
 
-  // Get ArgMax's next node(Cast)'s outputdefs
-  auto it = node.OutputEdgesBegin();
-  const auto* succ_node(graph_viewer.GetNode(it->GetNode().Index()));
+  // There are two cases here:
+  // 1. Special Case (ArgMax-Cast), we fuse the Argmax's output/Cast's input
+  // (We still have this special case here because CoreML model does not have Cast)
+  // 2. Otherwise, we add Argmax layer normally
+  if (node.GetOutputEdgesCount() == 1) {
+    auto it = node.OutputEdgesBegin();
+    const auto* succ_node(graph_viewer.GetNode(it->GetNode().Index()));
+    if (succ_node->OpType() == "Cast") {
+      // Skip the cast's input/argmax's output
+      *layer->mutable_input()->Add() = node.InputDefs()[0]->Name();
+      *layer->mutable_output()->Add() = succ_node->OutputDefs()[0]->Name();
+      model_builder.AddLayer(std::move(layer));
+      return Status::OK();
+    }
```

Comment:
If we do this we don't need the Cast node right? What makes the rest of the graph processing aware that the Cast op has been handled and can be ignored?

Reply:
oh first of all, this is based on the previous ep partitioning logic (might need to be changed).

What we do here is no different than the other operators. In`cast_op_builder`-`isopsupportedimpl`, we tell the ep that `cast`node is actually a "supported" operator and added it to the modelbuilder (but not actually added because we left it to be blank as coreml has no cast layer, info msg added for this part) So at this point, the graph processing should have a partition with ArgMax + Cast node and  be aware that cast op has been processed as a supported op. And here in `argmax_op_builder`, we then fuse the cast input with argmax's output and only add one argmax layer to the coreml model.
In other words, the rest of the graph should regard cast node as a "normal supported" operator and mark it as processed and partitioned.